### PR TITLE
Fix REPL.fuzzyscore() function using Levenshtein distance and converting to a range from 0 to 1

### LINKED ISSUE #49466: REPL.fuzzyscore heuristics could be better
The heuristics don't work very well in practise, partly because a mismatch near the beginning of the string completely messes up the algorithm, whereas numerous insertions/deletions don't do much.

For example
```julia
julia> REPL.fuzzyscore("bupercalifragilisticexpialidocious", "supercalifragilisticexpialidocious")
-68.0

julia> REPL.fuzzyscore("parasid", "supercalifragilisticexpialidocious")
6.59
```
Or
```julia
julia> REPL.fuzzyscore("IJilia", "IJulia")
-3.0933333333333337

julia> REPL.fuzzyscore("IJilia", "MotionCaptureJointCalibration")
5.5633333333333335
```
Maybe it would be better to use something like Lehvenstein by default with some slight heuristics on top?

### PR DESCRIPTION
**Pull request by PhoenixFlame101**
Closes #49466

 - Implements `fuzzyscore()` using `levenshtein()`, and calculates the score by limiting it to a range from 0 to 1.
 - Works similar to `stringsimilarity()` as described [here](https://github.com/JuliaLang/julia/issues/49466#issuecomment-1525785061).
 
Here's a few examples:

    julia> REPL.fuzzyscore("abc", "xyz")
    0.0

    julia> REPL.fuzzyscore("abc", "abc")
    1.0

    julia> REPL.fuzzyscore("bupercalifragilisticexpialidocious", "supercalifragilisticexpialidocious")
    0.9705882352941176

    julia> REPL.fuzzyscore("IJilia", "MotionCaptureJointCalibration")
    0.1724137931034483

### CODE REVIEW

**Comment by jakobnissen** [comment]
Thanks for the PR! There are some test failures that look legit. It also looks like `avgdistance` is not used elsewhere. Can it be deleted?

**Comment by jakobnissen** [comment]
The test failure appears to be that the `printmatches` is used, when the user types a help prompt in the REPL, to check whether the query matches multiple symbols. For example:
```julia
help?> !
search: ! != !== sum! put! pop! map! get! any! all!
```

And this is done with `fuzzyscore(word, match) < 0 && break`, but of course this never happens anymore. So, you need a new heuristic to determine when something is so far away that it's not relevant for users.

**Comment by KristofferC** [comment]
Still looks like some CI errors:

```
REPL                                             (13) \|         failed at 2023-05-03T13:00:03.048
--
Test Failed at /cache/build/default-amdci4-7/julialang/julia-master/julia-c15110f193/share/julia/stdlib/v1.10/REPL/test/docview.jl:34
Expression: occursin("'mutable struct'", str)
Evaluated: occursin("'mutable struct'", "search:\n\nCouldn't find 'mutable s'\nPerhaps you meant ismutable\n")
```

**Comment by tecosaur** [comment]
It's probably worth noting that levenshtein isn't great at transposition typos, e.g.
```julia
julia> 1 - REPL.levenshtein("hello", "helllo") / 6
0.8333333333333334

julia> 1 - REPL.levenshtein("hello", "ehllo") / 5
0.6
```

File: stdlib/REPL/test/docview.jl [hunk 1]
```diff
@@ -46,14 +46,19 @@ end
...
 end
 @testset "fuzzy score" begin
     # https://github.com/JunoLab/FuzzyCompletions.jl/issues/7
     # shouldn't throw when there is a space in a middle of query
     @test (REPL.matchinds("a ", "a file.txt"); true)
+
+    @test REPL.fuzzyscore("abc", "xyz") == 0
+    @test REPL.fuzzyscore("abc", "abc") == 1
+    @test REPL.fuzzyscore("abc", "dcab") == 0.25
+    @test REPL.fuzzyscore("abc", "dab") ≈ 1 / 3
```
Comment (LilithHafner) [inline stdlib/REPL/test/docview.jl:61]: These tests seem fragile to me. I would prefer to see
```suggestion
    @test REPL.fuzzyscore("IJilia", "IJulia") >
          REPL.fuzzyscore("IJilia", "MotionCaptureJointCalibration")
    @test REPL.fuzzyscore("bupercalifragilisticexpialidocious", "supercalifragilisticexpialidocious") >
          REPL.fuzzyscore("parasid", "supercalifragilisticexpialidocious")
```
From @jakobnissen's examples in #49466

**Comment by jakobnissen** [comment]
I'm going to close this in favor of #50412 . Nonetheless, thanks for giving it a go, @PhoenixFlame101 !

**Pull request closed without merging**